compaction/compaction.cc

```diff
@@ -746,6 +746,45 @@ class compaction {
         return consumer(make_compacting_reader(setup_sstable_reader(), compaction_time, max_purgeable_func(), gc_state));
     }
 
+    // when compacting a run of mutation fragments, the producer reads the
+    // mutations from sstables (or yet another reader). and the consumer
+    // processes the mutations, by either compacting them, or segregates them
+    // into chunks. each chunk of the fragments stream can be either piped to
+    // a sstable writer or yet another consumer. the consumers and writers can
+    // be created on demand.
+    //
+    //   def sstable_reader(sstables):
+    //       for sstable in sstables:
+    //           for frag in read(sstable):
+    //               yield frag
+    //
+    //   def compacting_reader(frag_gen):
+    //       for frag in compact(frag_gen):
+    //           yield frag
```

Comment:
Compaction might drop fragments or even entire partitions altogether.

Reply:
thank you @denesb . i will add this as a comment in the pseudocode code. the goal of these lists of pseudo code is to explain how these pipes are connected, and what their interfaces look like at a higher level. and yeah, would be a lot better if we can accompany them with accurate explanation regarding their behavior.